cluster-autoscaler/simulator/predicates.go

```diff
@@ -107,82 +105,64 @@ func NewPredicateChecker(kubeClient kube_client.Interface, stop <-chan struct{})
 	staticInitIfNeeded()
 
 	informerFactory := informers.NewSharedInformerFactory(kubeClient, 0)
-	podInformer := informerFactory.Core().V1().Pods()
+	providerRegistry := algorithmprovider.NewRegistry(1)
+	config := providerRegistry[scheduler_apis_config.SchedulerDefaultProviderName]
+	snapshot := scheduler_snapshot.NewEmptySnapshot()
 
-	sched, err := scheduler.New(
+	volumeBinder := scheduler_volumebinder.NewVolumeBinder(
```

Comment:
I wonder if we shouldn't give this some kind of fake kubeClient? It already has all the informers, it only needs kubeClient for mutating API calls. Obviously it shouldn't happen anyway, since we're only doing the filtering step and not a binding step, but it still feels safer not to give it a real client at all. WDYT?

Reply:
Ping

Reply:
Based on input from @ahg-g we need to pass real kubeclient here.